README_CI.md

```diff
@@ -0,0 +1,109 @@
+# Jenkins CI Architecture
+
+<!-- Note: This diagram can be re-regenerated by editing img/README_CI.puml.
+The VS Code PlantUML extension provides a good live-preview experience. -->
+![CI Architecture](./img/ci_architecture.svg)
```

Comment:
This is such a big enough diagram that it puts the ToC below the fold so I think I would prefer if the ToC came first.